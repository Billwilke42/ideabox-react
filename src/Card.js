import React from 'react';

const Card = ({ title, body, id, removeIdea, isFavorite }) => {
  const classString = `Card ${isFavorite && 'favorite'}`;
  return(
    <div className={classString}>
      <h1>{title}</h1>
      <p>{body}</p>
      <button>favorite</button>
      <button className='btn-remove' onClick={() => removeIdea(id)}>X</button>
    </div>
  );
}

export default Card;
