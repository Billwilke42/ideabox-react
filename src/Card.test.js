import React from 'react';
import {shallow, mount} from 'enzyme';
import Card from './Card';

describe('Card', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(
      <Card
        title="Bananas"
        body="brown and spotty"
        removeCard={jest.fn()}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should have a className of favorite if it is a favorite', () => {
    const wrapper = shallow(<Card title='burritos' isFavorite />);
    expect(wrapper.is('.favorite')).toEqual(true);
  });

  it('should call the removeIdea prop when clicked', () => {
    const removeIdeaMock = jest.fn();

    const wrapper = mount(
      <Card
        title='blah'
        body='hello'
        removeIdea={removeIdeaMock}
      />
    );
    wrapper.find('.btn-remove').simulate('click');
    expect(removeIdeaMock).toBeCalled();
  });
});
