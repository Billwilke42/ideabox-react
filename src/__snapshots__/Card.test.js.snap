// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    body="brown and spotty"
    removeCard={[MockFunction]}
    title="Bananas"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Bananas
        </h1>,
        <p>
          brown and spotty
        </p>,
        <button>
          favorite
        </button>,
        <button
          className="btn-remove"
          onClick={[Function]}
        >
          X
        </button>,
      ],
      "className": "Card undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bananas",
        },
        "ref": null,
        "rendered": "Bananas",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "brown and spotty",
        },
        "ref": null,
        "rendered": "brown and spotty",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "favorite",
        },
        "ref": null,
        "rendered": "favorite",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "btn-remove",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Bananas
          </h1>,
          <p>
            brown and spotty
          </p>,
          <button>
            favorite
          </button>,
          <button
            className="btn-remove"
            onClick={[Function]}
          >
            X
          </button>,
        ],
        "className": "Card undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bananas",
          },
          "ref": null,
          "rendered": "Bananas",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "brown and spotty",
          },
          "ref": null,
          "rendered": "brown and spotty",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "favorite",
          },
          "ref": null,
          "rendered": "favorite",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "btn-remove",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
